travis/format.sh

```diff
@@ -56,7 +56,7 @@ FILES_TO_CHECK="$(git diff $DIFF_OPTS -- master $FILETYPES)"
 
 for f in $FILES_TO_CHECK; do
   set +e
-  TRAILING_SPACES="$(grep -P " +$" $f)"
+  TRAILING_SPACES="$(grep '\s\+$' $f)"
```

Comment:
Note that `\s` also matches tab, return, vertical tab, form feed and, technically, newline.  I'd just use `[[:blank:]]`, since that just matches actual space and tab characters, and the others, while we probably want to remove them if they're in the code, it's more likely that they're intentionally there as part of a """ string.

Reply:
Oh, also `\s` (which is a synonym for `[[:space:]]`) is locale dependent and will match any whitespace characters in the locale, and those are even more likely to be intentional, IMO.

Reply:
But if we're sure we want to eliminate ALL whitespace at EOL, then I'm fine with `\s`, since there are workarounds if you really want whitespace at EOL in a """ string anyhow (which now that I think about it more is probably a bad practice in any case).

Reply:
Yeah, ignore me.  Listen to Chris. :-)